programs/bubblegum/program/src/lib.rs

```diff
@@ -888,6 +949,192 @@ fn process_collection_verification<'info>(
+    collection_metadata: &TokenMetadata,
+    token_metadata_program: &Program<'info, MplTokenMetadata>,
+) -> Result<()> {
+    // Mint account must match Collection mint
+    require!(
+        collection_mint.key() == collection.key,
+        BubblegumError::CollectionMismatch
+    );
+    // Verify correct account ownerships.
+    require!(
+        *collection_metadata_account_info.owner == token_metadata_program.key(),
+        BubblegumError::IncorrectOwner
+    );
+    // Collection mint must be owned by SPL token
+    require!(
+        *collection_mint.owner == spl_token::id(),
+        BubblegumError::IncorrectOwner
+    );
+
+    let collection_authority_record = match &collection_authority_record_pda {
+        None => None,
+        Some(authority_record_pda) => Some(authority_record_pda.to_account_info()),
+    };
+
+    // Assert that the correct Collection Authority was provided using token-metadata
+    assert_has_collection_authority(
+        &collection_metadata,
+        collection_mint.key,
+        collection_authority.key,
+        collection_authority_record.as_ref(),
+    )?;
+
+    Ok(())
+}
+
+fn process_update_metadata<'info, 'a>(
+    merkle_tree: &AccountInfo<'info>,
+    owner: &AccountInfo<'info>,
+    delegate: &AccountInfo<'info>,
+    compression_program: &AccountInfo<'info>,
+    tree_authority: &AccountInfo<'info>,
+    tree_authority_bump: u8,
+    log_wrapper: &Program<'info, Noop>,
+    remaining_accounts: &'a [AccountInfo<'info>],
+    root: [u8; 32],
+    old_metadata: MetadataArgs,
+    new_name: Option<String>,
+    new_symbol: Option<String>,
+    new_uri: Option<String>,
+    new_creators: Option<Vec<Creator>>,
+    new_seller_fee_basis_points: Option<u16>,
+    new_primary_sale_happened: Option<bool>,
+    new_is_mutable: Option<bool>,
+    nonce: u64,
+    index: u32,
+) -> Result<()> {
+    // Old metadata must be mutable to allow metadata update
+    require!(old_metadata.is_mutable, BubblegumError::MetadataImmutable);
+
+    let old_data_hash = hash_metadata(&old_metadata)?;
+    let old_creator_hash = hash_creators(&old_metadata.creators)?;
+
+    // Update metadata
+    let mut new_metadata = old_metadata;
+    if let Some(name) = new_name {
+        new_metadata.name = name;
+    };
+    if let Some(symbol) = new_symbol {
+        new_metadata.symbol = symbol;
+    };
+    if let Some(uri) = new_uri {
+        new_metadata.uri = uri;
+    };
+    if let Some(creators) = new_creators {
+        let old_creators = new_metadata.creators;
+        let no_new_creators_were_verified = creators
+            .iter()
+            .filter(|c| c.verified) // select only creators that are verified
+            .all(|c| {
+                old_creators
+                    .iter()
+                    .any(|old| old.address == c.address && old.verified)
+            });
+        require!(
+            no_new_creators_were_verified,
+            BubblegumError::CreatorDidNotVerify
+        );
```

Comment:
Yeah this check doesn't prevent the issuer of the update from un-verifying some creators. I can change the check to do that. Originally, I made the choice intentionally, since the `tree_delegate` has to sign-off on mints, I figured they could arbitrarily control `creators` for an NFT as desired, and then those `creators` can opt to sign and verify themselves or not. Under that premise, I thought it may be reasonable that the updater could unverify or even completely remove creators, if they desired (that is also permitted here).

Reply:
It does seem possible though that two parties could agree for a specific address to be a creator for some period of time, and then want to remove them afterwards. I.e. two parties agree that A will be a creator (and receive royalties) for 3 months, after which they will be removed. I suppose their `share` could also be updated to 0, but they may want to replace that creator slot.

Reply:
Let me know what you think about the above. I'll leave the check as-is for now, and only add a test that a new creator cannot be verified through this ix. But if after reading the thread above, you feel we still want to enforce the old creators cannot be removed/unverified I can add additional tests for that

Reply:
The current behaviour for NFTs is that `Update` does not remove verified creators nor unverify them. This prevents the authority to unilaterally remove a verified creator from the creator array, although as you pointed out, they can set the share to `0`. The only exception is when the authority itself is a creator.

Reply:
Added code that prevents removing existing verified creators during metadata update.

Note: I added the exception cases for the transaction signer as well (tree delegate): the tree delegate can add itself as a new verified creator, or remove/unverify itself even if it is a current verified creator.